sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuReadParquetFileFormat.scala

```diff
@@ -57,7 +58,19 @@ class GpuReadParquetFileFormat extends ParquetFileFormat with GpuReadFileFormatW
 }
 
 object GpuReadParquetFileFormat {
+
+  // copied from src/main/scala/org/apache/spark/sql/catalyst/util/package.scala
+  private val METADATA_COL_ATTR_KEY = "__metadata_col"
```

Comment:
I'd personally rather lean more into leveraging what Spark is doing to identify the metadata columns.  Rather than copying the logic, and hoping it stays in sync, we can let Spark identify the metadata columns for us.  When a metadata column is used, it shows up as a special type of attribute, `MetadataAttribute`.  Similar to what is done in 3.3's `FileSourceScanExec`, we could check if any the `output` attributes in `FileSourceScanExec` are actually `MetadataAttribute`.  If so, we avoid converting it to GPU.  Converting from `FileSourceScanExec` is already shimmed, so this would be a matter of putting this logic in the 3.3 shims for execs.

This also has the benefit of applying to _any_ file input format that is doing this, not just Parquet.  ORC, CSV, etc. aren't being handled in this PR currently, but ultimately they need to be.